app/src/app.rs

```diff
@@ -159,7 +192,33 @@ impl Component for App {
                 true
             }
             Msg::CollectStatus((slug, status)) => {
-                self.candidates_status.insert(slug.clone(), status.clone());
+                self.candidates_selection
+                    .insert(slug.to_string().clone(), status.clone());
+
+                if let Some(storage) = &mut self.local_storage {
+                    use yew::format::Json;
+                    storage.store("candidates-selection", Json(&self.candidates_selection));
+
+                    match storage.restore::<Json<Result<HashMap<String, CandidateStatus>, _>>>(
+                        "candidates-selection",
+                    ) {
+                        Json(Ok(x)) => {
+                            crate::log!("Local storage: {:?}", x)
+                        }
+                        Json(Err(_)) => {
+                            crate::log!("Cannot restore data from local storage")
+                        }
+                    }
+                }
+                true
+            }
+            Msg::ClearSelection => {
+                if let Some(storage) = &mut self.local_storage {
+                    storage.remove("candidates-selection");
+                    crate::log!("local storage cleared")
+                }
+                self.candidates_selection.clear();
+                crate::log!("selection cleared");
```

Comment:
Unnecessary logs again. Try to use logs to convey a message that the developer or the user can not know because not obvious.
```suggestion
                if let Some(storage) = &mut self.local_storage {
                    storage.remove("candidates-selection");
                }
                self.candidates_selection.clear();
```